pkg/services/govmomi/clustermodules/provider.go

```diff
@@ -87,19 +88,18 @@ func (cm *provider) DoesModuleExist(ctx context.Context, moduleID string, cluste
 		return false, nil
 	}
 
-	modules, err := cm.manager.ListModules(ctx)
-	if err != nil {
-		return false, err
+	_, err := cm.manager.ListModuleMembers(ctx, moduleID)
+	if err == nil {
+		return true, nil
 	}
 
-	for _, mod := range modules {
-		if mod.Cluster == clusterRef.Value && mod.Module == moduleID {
-			return true, nil
-		}
+	// Is there a better way of unwrap vAPI errors?
+	if strings.Contains(err.Error(), "404 Not Found") {
```

Comment:
Not sure if that actually is a better variant.
We can't do a good type assertion because the error type is a private one

```suggestion
	if strings.Contains(err.Error(), fmt.Sprintf("%d %s", http.StatusNotFound, http.StatusText(http.StatusNotFound))) {
```